M2.el

```diff
@@ -423,7 +423,8 @@ SEND-TO-BUFFER."
   (M2--send-to-program-helper send-to-buffer
 			      (save-excursion (M2-to-end-of-prompt) (point))
 			      (line-end-position))
-  (forward-line))
+  (forward-line)
+  (when (and (eobp) (not (bolp))) (newline)))
```

Comment:
Could you maybe leave a comment on this line that says what it does? "and eobp not bolp" sounds like gobbledygook tbh XD

Reply:
Sure thing!